# HG changeset patch
# User goodwillcoding <goodwillcoding@webhippo.net>
# Date 1376769053 25200
# Node ID c31841d2fd2171419134f30e815dcafb1e8c7eee
# Parent  7304e4759ae65343d89a51359ca538912519cc31
Make pygments.lexers.guess_lexer_for_filename py3 compatible.
Background:
In py3 one can longer sort classes by class names
the fix addes a custome sorting function that works
the same in py2 and py3

diff --git a/pygments/lexers/__init__.py b/pygments/lexers/__init__.py
--- a/pygments/lexers/__init__.py
+++ b/pygments/lexers/__init__.py
@@ -188,7 +188,13 @@
         if rv == 1.0:
             return lexer(**options)
         result.append((rv, lexer))
-    result.sort()
+
+    # since py3 can no longer sort by class name by default, here is the
+    # sorting function that works in both
+    def type_sort(type_):
+        return (type_[0], type_[1].__name__)
+    result.sort(key=type_sort)
+
     if not result[-1][0] and primary is not None:
         return primary(**options)
     return result[-1][1](**options)
